src/Adapter/Preferences/PreferencesConfiguration.php

```diff
@@ -104,6 +116,12 @@ public function updateConfiguration(array $configuration)
         $this->configuration->set('PS_MULTISHOP_FEATURE_ACTIVE', $configuration['multishop_feature_active']);
         $this->configuration->set('PS_SHOP_ACTIVITY', $configuration['shop_activity']);
 
+        // Update product page feature automatically based on PS_MULTISHOP_FEATURE_ACTIVE
+        if (!$previousMultistoreFeatureState && (bool) $this->configuration->get('PS_MULTISHOP_FEATURE_ACTIVE')) {
+            $this->featureFlagRepository->disable(FeatureFlagSettings::FEATURE_FLAG_PRODUCT_PAGE_V2);
```

Comment:
Didn't we say we shouldn't disable this one? Since anyway the `FEATURE_FLAG_PRODUCT_PAGE_V2` is not used when we are in multishop.

However, if you change the setting by mistake and reset it just after in the meantime the product v2 has been disabled and you switch to the v1 for no reason (the multishop is not really active until you created a second shop).